sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -563,9 +568,29 @@ protected void beforeCreating() {
                 .withNetworkSecurityGroup(new NetworkSecurityGroupInner().withId(networkSecurityGroup.id()));
         }
 
-        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values());
+        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values(), this.specifiedIpConfigNames);
 
         // Reset and update IP configs
         this.innerModel().withIpConfigurations(innersFromWrappers(this.nicIPConfigurations.values()));
     }
+
+    @Override
+    public NetworkInterfaceImpl withPrimaryPublicIPAddressDeleteOptions(DeleteOptions deleteOptions) {
+        this.ensureDeleteOptions(deleteOptions, "primary");
+        return this;
+    }
+
+    @Override
+    public NetworkInterfaceImpl update() {
+        this.specifiedIpConfigNames = new HashMap<String, DeleteOptions>();
+        return super.update();
+    }
+
+    public void ensureDeleteOptions(DeleteOptions deleteOptions, String ipConfigName) {
+        if (this.specifiedIpConfigNames.containsKey(ipConfigName)) {
+            this.specifiedIpConfigNames.replace(ipConfigName, deleteOptions);
+        } else {
+            this.specifiedIpConfigNames.put(ipConfigName, deleteOptions);
+        }
```

Comment:
What is the difference between this code, with `this.specifiedIpConfigNames.put(ipConfigName, deleteOptions);`?

Reply:
It has been simplified to 
```
    public void ensureDeleteOptions(DeleteOptions deleteOptions, String ipConfigName) {
        this.specifiedIpConfigNames.put(ipConfigName, deleteOptions);
    }
```
in the new version